### ISSUE TITLE
Dealing with data with resolutions which are not a power of 2?

### ISSUE THREAD

**Issue by dericknganyu** [gaurav71531/mwt-operator#3]
https://github.com/gaurav71531/mwt-operator/blob/4b0770c0987a325b5a18775a0bdb63ab6d69435e/models/models.py#L141

This can be seen at the start of each `wavelet_transform` function. 

For example, if  `x` has shape `(B, 65, 65, W)`, the concatenation returns an error, because on the second axis, the slice ::2 and 1::2 return different dimensions. 
```
xa = torch.cat([x[:, ::2, :, :],
                        x[:, 1::2, :, :],
                       ], -1)
```
Thanks for sharing these very nice results.

**Comment by gaurav71531** [comment]
The transform supports only powers of 2 due to the binary division and combination as you have seen already. For other shapes, the best solution is to zero-pad to the higher level of 2^n.
